src/Agent.Worker/Build/BuildDirectoryManager.cs

```diff
@@ -122,7 +122,14 @@ public sealed class BuildDirectoryManager : AgentService, IBuildDirectoryManager
             // Set the default clone path for each repository (the Checkout task may override this later)
             foreach (var repository in repositories)
             {
-                var repoPath = GetDefaultRepositoryPath(executionContext, repository, newConfig.SourcesDirectory);
+                var repoSourceDirectory = newConfig?.RepositoryTrackingInfo.Where(item => item.Identifier == repository.Alias).Select(item => item.SourcesDirectory).FirstOrDefault();
```

Comment:
Should there be case-insensitive comparison like in [row](https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Agent.Worker/Build/BuildDirectoryManager.cs#L165)

Reply:
Yeah, I think it make sense, thank you.

Reply:
Done